From 1d381356826ffb1184ae8cc69020b16da54d2c50 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:50:00 -0400
Subject: [fs] nfsd: use more generous NFS4_ACL_MAX

Message-id: <1403560314-32525-59-git-send-email-bfields@redhat.com>
Patchwork-id: 84479
O-Subject: [Patch RHEL7 058/172] nfsd4: use more generous NFS4_ACL_MAX
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: d139977d002e34d687e2ebc1157a234d32183465
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/acl.h b/fs/nfsd/acl.h
index afd3e0e..1dee324 100644
--- a/fs/nfsd/acl.h
+++ b/fs/nfsd/acl.h
@@ -37,9 +37,13 @@
 
 #include <linux/posix_acl.h>
 
-/* Maximum ACL we'll accept from client; chosen (somewhat arbitrarily) to
- * fit in a page: */
-#define NFS4_ACL_MAX 170
+/*
+ * Maximum ACL we'll accept from a client; chosen (somewhat
+ * arbitrarily) so that kmalloc'ing the ACL shouldn't require a
+ * high-order allocation.  This allows 204 ACEs on x86_64:
+ */
+#define NFS4_ACL_MAX ((PAGE_SIZE - sizeof(struct nfs4_acl)) \
+   / sizeof(struct nfs4_ace))
 
 struct nfs4_acl *nfs4_acl_new(int);
 int nfs4_acl_get_whotype(char *, u32);
-- 
1.7.1